shell/components/AppModal.vue

```diff
@@ -107,7 +111,10 @@ export default defineComponent({
         width: this.modalWidth,
         ...this.stylesPropToObj,
       };
-    }
+    },
+    isCodeMirrorFocused() {
```

Comment:
I think that we will be able to remove the need for `AppModal` to know about the code mirror if we solve the issue with event listeners on the document.

Reply:
Since the `onReady` event was already being bubbled up with the correct reference of the code mirror instance, I've updated `Import` to pass it to `Header` then we can use it in `AppModal`. With that we could remove all selectors and weird logic.